modules/core/src/opencl/convert.cl

```diff
@@ -68,7 +71,11 @@ __kernel void convertTo(__global const uchar * srcptr, int src_step, int src_off
             __global const srcT * src = (__global const srcT *)(srcptr + src_index);
             __global dstT * dst = (__global dstT *)(dstptr + dst_index);
 
+#ifdef NO_SCALE
+            dst[0] = src[0];
```

Comment:
`convertToDT` can include rounding policy (_rte, _sat) that may differ from the default type cast:
https://github.com/Itseez/opencv/blob/c0b544af705eb058642cd8387abf3d9536eaa32a/modules/core/src/ocl.cpp#L5585-L5603